instrumentation/src/javatest/org/oppia/android/instrumentation/ExplorationPlayerTest.kt

```diff
@@ -0,0 +1,108 @@
+import androidx.test.uiautomator.UiScrollable
+import androidx.test.uiautomator.UiSelector
+import androidx.test.uiautomator.Until.hasObject
+import org.junit.Assert.assertNotNull
+import org.junit.Before
+import org.junit.Test
+
+/** Tests for Explorations. */
+class ExplorationPlayerTest {
+  private val OPPIA_PACKAGE = "org.oppia.android"
+  private val LAUNCH_TIMEOUT = 30000L
+  private val TRANSITION_TIMEOUT = 5000L
+  private lateinit var device: UiDevice
+
+  @Before
+  fun startOppiaFromHomeScreen() {
+    // Initialize UiDevice instance
+    device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+
+    // Start from the home screen
+    device.pressHome()
+
+    // Wait for launcher
+    val launcherPackage = getLauncherPackageName()
+    assertNotNull(launcherPackage)
+    device.wait(hasObject(By.pkg(launcherPackage)), LAUNCH_TIMEOUT)
+
+    // Launch the blueprint app
+    val context = ApplicationProvider.getApplicationContext<Context>()
+    val intent = context.packageManager
+      .getLaunchIntentForPackage(OPPIA_PACKAGE)
+    intent!!.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK) // Clear out any previous instances
+    context.startActivity(intent)
+
+    // Wait for the app to appear
+    device.wait(hasObject(By.pkg(OPPIA_PACKAGE).depth(0)), LAUNCH_TIMEOUT)
+  }
+
+  @Test
+  fun testExploration_prototypeExploration_toolbarTitle_isDisplayedSuccessfully() {
+    NavigateToPrototypeExploration()
+    device.wait(
+      hasObject(
+        By.res("$OPPIA_PACKAGE:id/exploration_toolbar_title")
+      ),
+      TRANSITION_TIMEOUT
+    )
+    assertNotNull(device.findObject(By.res("$OPPIA_PACKAGE:id/exploration_toolbar_title")))
+  }
+
+  /** Navigates and opens the Prototype Exploration using the admin profile. */
+  fun NavigateToPrototypeExploration() {
+    val skip_button = device.findObject(By.res("$OPPIA_PACKAGE:id/skip_text_view"))
+    skip_button?.let {
+      it.click()
+      device.wait(
+        hasObject(By.res("$OPPIA_PACKAGE:id/get_started_button")),
+        TRANSITION_TIMEOUT
+      )
+      device.findObject(By.res("$OPPIA_PACKAGE:id/get_started_button"))
+        .click()
+    }
+    device.wait(
+      hasObject(By.res("$OPPIA_PACKAGE:id/profile_select_text")),
+      TRANSITION_TIMEOUT
+    )
+    val profiles = UiCollection(UiSelector().className("androidx.recyclerview.widget.RecyclerView"))
+    profiles.getChildByText(UiSelector().className("android.widget.LinearLayout"), "Admin").click()
+    val recyclerview = UiScrollable(UiSelector().scrollable(true))
+    recyclerview.setAsVerticalList()
+    recyclerview.scrollTextIntoView("First Test Topic")
+    val firstTestTopicText = device.findObject(UiSelector().text("First Test Topic"))
+    firstTestTopicText.click()
+    device.findObject(UiSelector().text("First Test Topic"))
+      .click()
+    device.findObject(UiSelector().text("LESSONS"))
+      .click()
+    device.findObject(UiSelector().text("First Story"))
+      .click()
```

Comment:
These can also fit on their own lines--pleas eupdate. Ditto for the click() above, too.

Reply:
Done